src/atoms/firebaseUserAtoms.ts

```diff
@@ -79,9 +108,8 @@ export const signInWithGoogleAtom = atom(null, (get, set, _) => {
       .catch(error => {
         if (error.code === 'auth/credential-already-in-use') {
           // user already has an account. Sign in to that account and override our data.
-          confirmOverrideData(() => {
-            firebase.auth().signInWithCredential(error.credential);
-          });
+          set(CredentialforDataOverrideAtom, error.credential);
+          set(showConfirmModal, true);
```

Comment:
Just set a singular atom here, passing in a callback for what happens when the user confirms data override. ie:

```
firebase.auth().signInWithCredential(error.credential);
```